### PR TITLE
Add date function to get current date without time component

### PR DESCRIPTION
**Pull request by paulrutter**
- Add CURDATE/CURRENT_DATE method to AlaSQL
- Add test

This removed the need for:

```
DATE(YEAR(NOW()) + '-' + MONTH(NOW()) + '-' + DAY(NOW()))
```

Thank you for the time you are putting into AlaSQL!

### CODE REVIEW

**Comment by paulrutter** [comment]
@mathiasrw any feedback on this pr? Thanks!

**Comment by mathiasrw** [comment]
Interesting, relevant and missing. Nice. 

    SELECT CURDATE();
    
looks good to me. I do feel like we should avoid none standard notation like `SELECT CURRENT_DATE();` and instead support the widely use same notation as MySQL and MariaDB with `SELECT CURRENT_DATE;` where its not a function, but a constant. 

Are you feeling the mojo to try to dive into that? Adding it to src/alasqlparser.jison should not be too complex, but I admit its a bit exotic if you have not worked with parsers before.

**Comment by paulrutter** [comment]
Isn't the function exactly what MySQL offers?
See https://www.w3schools.com/sql/func_mysql_curdate.asp and https://www.w3schools.com/sql/func_mysql_current_date.asp and https://www.techonthenet.com/mariadb/functions/current_date.php#:~:text=The%20CURRENT_DATE%20function%20will%20return,synonym%20for%20the%20CURDATE%20function.

I do see that most also support an alias as constant, although we don't offer this for other date functions except CURRENT_TIMESTAMP in alasql. The current approach would be in line with other AlaSQL date functions. Do you want me to look into using these aliases? 

To the parser point, i'm not seeing the other date functions in there. Is that deliberate? Only CURRENT_TIMESTAMP is there, not NOW for example.

I can look into adding it to the parser though next week.

**Comment by paulrutter** [comment]
@mathiasrw I tried rebuilding the parser via `npm run jison`. but the resulting alasqlparser.js is completely different from what's in git right now. Should i use a different command?

I wouldn't expect changes against what's in git.
Also, when i run the tests after generating the parser again, a lot of tests fail due to the parser throwing errors.

I found https://github.com/AlaSQL/alasql/pull/1903#issuecomment-2004332511, will try that and update the package.json if that works.

**Comment by paulrutter** [comment]
@mathiasrw 
Can you take a look at the changes made?
I also added a section to rebuild the parser and added both CURDATE as CURRENT_DATE.
It can be used with and without parenthesis.

It feels to me like other date functions should be added to the parser as well, as these are currently missing.
`DATE`/`NOW`/`GETDATE`/`YEAR`/`MONTH` for example.

**Comment by paulrutter** [comment]
https://github.com/AlaSQL/alasql/issues/1936

File: README.md [hunk 1]
```diff
@@ -629,6 +629,15 @@ AlaSQL is an [OPEN Open Source Project](http://openopensource.org/). This means
...
 
+## Rebuilding the parser 
+
+To rebuild the parser, follow these steps:
+
+* Make changes to alasqlparser.jison
+* `npm install -g jison`
+* `npm run jison`
+* `npm test` to validate the changes made
+* Commit changes to alasqlparser.jison and alasqlparser.js
```
Comment (mathiasrw) [inline README.md:640]: Thank you!

File: package.json [hunk 2]
```diff
@@ -62,7 +62,7 @@
 		"eslint-plugin-promise": "6.1.1",
 		"git-branch-is": "4.0.0",
 		"husky": "8.0.3",
-		"jison-gho": "^0.6.1-216",
```
Comment (mathiasrw) [inline package.json:65]: Good to see you dived in. There was a security issue with a subdependency - that is why we transitioned to -gho. Looks like its fixed now.

File: package.json [hunk 3]
```diff
@@ -15,15 +15,15 @@
 	"scripts": {
 		"test": "sh build.sh && yarn test-only",
 		"test-ci": "(yarn test-format || 1) && yarn test && yarn install-g && alasql 'select 1 as Succes'",
-		"test-only": "mocha ./test --reporter dot # npx bun node_modules/.bin/mocha ./test --reporter dot",
+		"test-only": "node node_modules/mocha/bin/mocha.js ./test --reporter dot",
 		"#test-only": "(command -v bun && bun node_modules/.bin/mocha ./test --reporter dot) || npx bun node_modules/.bin/mocha ./test --reporter dot",
```
Comment (mathiasrw) [inline package.json:19]: Thanks!

**Review by mathiasrw — APPROVED** [review]

**Comment by paulrutter** [comment]
Thanks for merging, can we expect a new release soon?
@mathiasrw

**Comment by mathiasrw** [comment]
Released as part of v4.5.0

**Pull request merged**